src/utils/rateLimiter.util.ts

```diff
@@ -0,0 +1,27 @@
+import { RedisService } from '@liaoliaots/nestjs-redis';
+import { Injectable, Logger } from '@nestjs/common';
+
+@Injectable()
+export class RateLimiterService {
+  private readonly logger = new Logger(RateLimiterService.name);
+
+  constructor(private readonly redisService: RedisService) {}
+
+  async isRateLimited(senderId: string, repoId: string): Promise<boolean> {
+    const client = this.redisService.getClient();
+    const key = `rate_limit:${senderId}:${repoId}`;
+    const currentCount = await client.get(key);
+
+    if (currentCount && parseInt(currentCount) >= 5) {
+      return true;
+    }
```

Comment:
PullProofAI

Consider adding error handling for Redis operations and parsing integers. It's advisable to handle potential errors that may occur during Redis operations or when converting `currentCount` to an integer.
```diff
+     if (isNaN(parseInt(currentCount))) {
+         throw new Error('Failed to parse currentCount to integer');
+     }
```



<!-- This is an auto-generated comment by PullProofAI -->